src/Resolver/Factory.php

```diff
@@ -19,8 +19,20 @@
 use React\EventLoop\Loop;
 use React\EventLoop\LoopInterface;
 
-final class Factory
+class Factory
```

Comment:
What is the motivation for this change? Seems to be unrelated to the PR title?

Not sure I like this without having a better understanding of the motivation, see also #134.